### PR TITLE
Fix segment data disappears from UI after changing auto_archive setting 

### LINKED ISSUE #10036: Segment data disappears from UI after changing auto_archive setting
Segment data disappears from UI after changing auto_archive setting from 0 to 1. After changing back, data appears again. Version 2.16. Please see attached video:

https://drive.google.com/a/piwik.pro/file/d/0B2rhGHbcCX73dmMtSUJUdHRBQzQ/view?usp=sharing

One has to be logged in with piwik.pro credentials to view it.

FYI @mattab

### PR DESCRIPTION
**Pull request by tsteur**
refs #10036

This should fix the problem. @quba or @mgonera can you try to apply the patch and check if it's fixed?

### CODE REVIEW

**Comment by mattab** [comment]
This was a tricky one to find - well done @tsteur - looking forward to feedback and hopefully this fixes the issue fully.

**Comment by mgonera** [comment]
@tsteur we applied the fix but it didn't solve the issue. what more can we deliver?

**Comment by tsteur** [comment]
You might have to wait for new archives. When was the patch applied? Are you testing with today? I remember you and @quba described a different issue but we thought it would solve both issues. It's possible there's still another issue or maybe it works with today or when the cron runs the next time or so.

**Comment by mgonera** [comment]
@tsteur so you mean this will be solved only in the future compiled archives?

Please take a look at the video provided in #10036 :

https://drive.google.com/a/piwik.pro/file/d/0B2rhGHbcCX73dmMtSUJUdHRBQzQ/view?usp=sharing

This issue still persists without any change.

@quba I cannot understand whether what you described would help in this case? Please let me know

**Comment by mattab** [comment]
Hi @mgonera @quba 
did you have a chance to confirm this fix works as expected?

**Comment by mgonera** [comment]
@mattab as I said earlier:

> This issue still persists without any change.

we are currently investigating root cause on our own as it is impacting more and more environments.

**Comment by mattab** [comment]
thanks for letting me know!

File: core/Archive.php [hunk 1]
```diff
@@ -733,6 +738,8 @@ private function cacheArchiveIdsWithoutLaunching($plugins)
         foreach ($plugins as $plugin) {
             $doneFlag = $this->getDoneStringForPlugin($plugin, $this->params->getIdSites());
             $this->initializeArchiveIdCache($doneFlag);
+            $globalDoneFlag = Rules::getDoneFlagArchiveContainsAllPlugins($this->params->getSegment());
+            $this->initializeArchiveIdCache($globalDoneFlag);
```
Comment (mattab) [inline core/Archive.php]: @tsteur Did you notice there is also the call to `initializeArchiveIdCache` in `prepareArchive` method here: https://github.com/piwik/piwik/blob/2.16.1/core/Archive.php#L900-L901

should we add the initializeIdCache in this method as well?

Comment (tsteur) [inline core/Archive.php]: It was not needed there, I don't remember why anymore. We had a look that time.

**Pull request merged**